model/build-processor/src/main/java/org/keycloak/models/map/processor/AbstractGenerateEntityImplementationsProcessor.java

```diff
@@ -175,12 +176,16 @@ protected boolean isCollection(TypeMirror fieldType) {
     }
 
     protected String deepClone(TypeMirror fieldType, String parameterName) {
+        TypeElement typeElement = elements.getTypeElement(types.erasure(fieldType).toString());
         if (isKnownCollectionOfImmutableFinalTypes(fieldType)) {
-            TypeElement typeElement = elements.getTypeElement(types.erasure(fieldType).toString());
             return parameterName + " == null ? null : " + interfaceToImplementation(typeElement, parameterName);
-        } else {
-            return "deepClone(" + parameterName + ")";
+        } else if (isMapType(typeElement)) {
+            TypeMirror mapValueType = getGenericsDeclaration(fieldType).get(1);
+            if (!isImmutableFinalType(mapValueType)) {
+                return parameterName + " == null ? null : " + parameterName + ".entrySet().stream().collect(java.util.stream.Collectors.toMap(java.util.Map.Entry::getKey, entry -> " + deepClone(mapValueType, "entry.getValue()") + "))";
```

Comment:
We need to provide the constructor to construct appropriate map, there is no guarantee on the type or mutability of the resulting map; i.e. use [the four parameter `Collectors.toMap()` method](https://docs.oracle.com/javase/8/docs/api/java/util/stream/Collectors.html#toMap-java.util.function.Function-java.util.function.Function-java.util.function.BinaryOperator-java.util.function.Supplier-), where `mergeFunction` can be `(o1,o2)->o1` since there can be no collision, and `HashMap::new` would do for the time being.